firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java

```diff
@@ -220,9 +220,6 @@ private ViewSnapshot initializeViewAndComputeSnapshot(Query query, int targetId)
     View view = new View(query, queryResult.getRemoteKeys());
     View.DocumentChanges viewDocChanges = view.computeDocChanges(queryResult.getDocuments());
     ViewChange viewChange = view.applyChanges(viewDocChanges, synthesizedCurrentChange);
-    hardAssert(
```

Comment:
I believe the reason that this assert was here is that this method does not call `updateTrackedLimboDocuments`. 

So, we either need to convince ourselves that any limbo document encountered here will already be in limbo thanks to the originating query (and document this assumption) or call `updateTrackedLimboDocuments`.

Reply:
Added a call to updateTrackedLimboDocuments